openeo/rest/userfile.py

```diff
@@ -0,0 +1,58 @@
+import typing
+from typing import Any, Dict, List, Union
+from pathlib import Path
+
+if typing.TYPE_CHECKING:
+    # Imports for type checking only (circular import issue at runtime).
+    from openeo.rest.connection import Connection
+
+
+class UserFile:
+    """Represents a file in the user-workspace of openeo."""
+
+    def __init__(self, path: str, connection: 'Connection', metadata: Dict[str, Any] = {}):
+        self.path = path
+        self.metadata = metadata
```

Comment:
I'd make the argument `metadata` `None` by default (it's bad style to have mutable objects like dicts as default values) and fall back on only the path data:
```python
def __init__(self, path: str, connection: 'Connection', metadata: Optional[Dict[str, Any]] = None):
        self.path = path
        self.metadata = metadata or {"path": path}
```

that will simplify `to_dict` too

Reply:
Done